subprojects/docs/src/docs/userguide/javaTesting.adoc

```diff
@@ -0,0 +1,380 @@
+
+NOTE: Not all test frameworks play well with filtering. Some advanced, synthetic tests may not be fully compatible. However, the vast majority of tests and use cases work perfectly well with Gradle's filtering mechanism.
+
+The following two sections look at the specific cases of simple class/method names and fully-qualified names.
+
+[[simple_name_pattern]]
+==== Simple name pattern
+
+Since 4.7, Gradle has treated a pattern starting with an uppercase letter as a simple class name, or a class name + method name. For example, the following command lines run either all or exactly one of the tests in the `SomeTestClass` test case, regardless of what  package it's in:
+
+```
+# Executes all tests in SomeTestClass
+gradle test --tests SomeTestClass
+
+# Executes a single specified test in SomeTestClass
+gradle test --tests SomeTestClass.someSpecificMethod
+
+gradle test --tests SomeTestClass.*someMethod*
+```
+
+[[full_qualified_name_pattern]]
+==== Fully-qualified name pattern
+
+Prior to 4.7 or if the pattern doesn't start with an uppercase letter, Gradle treats the pattern as fully-qualified. So if you want to use the test class name irrespective of its package, you would use `--tests *.SomeTestClass`. Here are some more examples:
+
+```
+# specific class
+gradle test --tests org.gradle.SomeTestClass
+
+# specific class and method
+gradle test --tests org.gradle.SomeTestClass.someSpecificMethod
+
+# method name containing spaces
+gradle test --tests "org.gradle.SomeTestClass.some method containing spaces"
+
+# all classes at specific package (recursively)
+gradle test --tests 'all.in.specific.package*'
+
+# specific method at specific package (recursively)
+gradle test --tests 'all.in.specific.package*.someSpecificMethod'
+
+gradle test --tests '*IntegTest'
+
+gradle test --tests '*IntegTest*ui*'
+
+gradle test --tests '*ParameterizedTest.foo*'
+
+# the second iteration of a parameterized test
+gradle test --tests '*ParameterizedTest.*[2]'
+```
+
+Note that the wildcard '*' has no special understanding of the '.' package separator. It's purely text based. So `--tests *.SomeTestClass` will match any package, regardless of its 'depth'.
+
+You can also combine filters defined at the command line with <<sec:continuous_build, continuous build>> to re-execute a subset of tests immediately after every change to a production or test source file. The following executes all tests in the 'com.mypackage.foo' package or subpackages whenever a change triggers the tests to run:
+
+```
+gradle test --continuous --tests "com.mypackage.foo.*"
+```
+
+[[sec:single_test_execution_via_system_properties]]
+=== Single test execution via System Properties
```

Comment:
The main reason to leave it in is so that people know what it's about when they encounter it in the wild, for example on forums. I'd be happy to rework the section to explicitly explain that the option should no longer be used and is only covered for that reason. Make sense? Or do you think it should be removed regardless?

Reply:
Leaving a single sentence in for now would be okay, just not the full explanation.